docker/candlepin-base/cp-test

```diff
@@ -49,6 +49,29 @@ validate_translation() {
     fi
 }
 
+gettext() {
+    if [ -f ./gradlew ]; then
+        ./gradlew gettext
+    fi
+
+    #Code to commit the updated files
+    CHANGED_FILE=`git diff --name-only HEAD`
+    ADD_COUNT=0
+
+    for file in $(echo $CHANGED_FILE | tr "\n" "\n")
+    do
+      echo "Performing: git add $file."
+      git add $file
+      ((ADD_COUNT++))
+    done
+
+    if [ $ADD_COUNT -gt 0 ]; then
+      echo "Committing and pushing the above mentioned files."
+      git commit -m "keys.pot file is updated"
```

Comment:
In addition to the gettext changes, we should also be running msgattrib right after it in this job (which will be removing strings from the .po files, based on them being removed from the .pot file). That's the 'Replace the second zanata pull with' step from the design document.

For how that is done, we have a couple of options. In shell script it could be done as such:
```
for pofile in common/po/*.po; do
    msgattrib --set-obsolete --ignore-file=common/po/keys.pot -o $pofile $pofile;
    msgattrib --no-obsolete -o $pofile $pofile;
done;
```
But maybe a more consistent way would be to do add it as another gradle task in the Gettext.groovy file we have (similar to the existing gradle `gettext` and `msgmerge` commands we have).

Reply:
Agreed, raised a PR https://github.com/candlepin/candlepin/pull/2850 to add new gradlew task `msgattrib`.